javascripts/discourse/widgets/layouts-category-list.js

```diff
@@ -186,6 +190,54 @@ export default layouts.createLayoutsWidget('category-list', {
   }
 });
 
+createWidget('layouts-minimize-categories', {
+  tagName: 'li.layouts-minimize-button.layouts-category-link',
+  buildKey: (attrs) => 'layouts-minimize-categories',
+  
+  defaultState(attrs) {
+    return {
+      sidebarMinimized: false
+    }
+  },
+
+  html(attrs, state) {
+    let contents = [];
+    const minimizeIcon = h('div.category-logo.minimize-icon', iconNode('chevron-circle-left'));
+    const minimizeText = h('div.category-name.minimize-text', I18n.t(themePrefix("minimize_button_label")));
+
+    contents.push(minimizeIcon);
+    contents.push(minimizeText);
+
+    return contents;
+  },
+  
+  click(attrs) {
+    const categoryDetails = document.querySelectorAll('.layouts-category-link .category-name, .layouts-category-link .d-icon-lock');
+    const sidebar = document.querySelector('aside.sidebar');
+    const minimizeIcon = document.querySelector('.minimize-icon');
+    const siteSettings = this.register.lookup("site-settings:main");
+    const sidebarLeftWidth = siteSettings.layouts_sidebar_left_width;
+    
+    if (!this.state.sidebarMinimized) {
+      categoryDetails.forEach((category) => {
```

Comment:
@keegangeorge Like with jquery we should only be using javascript to change the classList outside of the render thread of the frameworks we're using if it's the last option available. Rather we should be using:

1. Events to send actions up the widget hierarchy
2. Attributes to send values and current state down the widget hiearchy

Then updating the classlists of various widgets or their html using ``attrs`` and ``state``. Only if that's not possible do we resort to updating the DOM directly like this. Same in Ember as well. Try to use the framework first and direct class manipulation like this as a last resort.

If you're not sure what I'm talking about here, let's chat about it in our catchup this week.